Formula/zrepl.rb

```diff
@@ -0,0 +1,27 @@
+class Zrepl < Formula
+  desc "One-stop ZFS backup & replication solution"
+  homepage "https://zrepl.github.io"
+  url "https://github.com/zrepl/zrepl/archive/v0.1.1.tar.gz"
+  sha256 "0c16554e4527d14a390d78cf95bce759da425019a83ec63acfed5b4c50d68c9c"
+  head "https://github.com/zrepl/zrepl.git"
+
+  depends_on "go" => :build
+
+  def install
+    contents = Dir["{*,.git,.gitignore}"]
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/zrepl/zrepl").install contents
+
+    ENV["GOPATH"] = gopath
+    ENV.prepend_create_path "PATH", gopath/"bin"    
+    cd gopath/"src/github.com/zrepl/zrepl" do
+      system "./lazy.sh",  "godep"
+      system "make", "ZREPL_VERSION=0.1.1"
+      bin.install "artifacts/zrepl"
+    end      
+  end
+
+  test do
+    system "#{bin}/zrepl", "help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
ok, I will check with @problame to see what might be a good test

Reply:
@fxcoudert Would a config check be sufficient? I still need to figure out how to ship a default config with it and how to integrate it in this formula, but would verifying the config be sufficient?

Reply:
would verifying the config be sufficient? znapzend (a tool very similar in scope and utility) also doesn't do much tests...

Reply:
Anything that actually enacts some part of the program instead of spitting out something static is a good enough test.

Reply:
I just added a test that downloads a sample config and checks the sample config.